docs/create-packages/Sign-a-Package.md

```diff
@@ -0,0 +1,101 @@
+
+Package repositories who implement repository signatures will provide a signed version of all packages, no matter if the package is author-signed or not. This capability allows to verify the complete dependency graph even for cases where a signed package has a dependency on an unsigned package. 
+
+Packages with the repository signature offer content integrity and proof of origin, even when the package is found in a different mirror/repository.
+
+# How to Author-Sign a package
+
+NuGet package authors who want to sign a package must obtain a X.509 code signing certificate. It’s strongly recommended to obtain the certificate from a public certification authority. The author can also use a certificate from a private certification authority but these signatures will generate a warning if the certificate is not trusted in the machine where the validation will occur.
+
+## Certificate Requirements
+
+Authors require a code signing certificate, these are a special type of certificate that are valid for the `id-kp-codeSigning` purpose [[RFC 5280 section 4.2.1.12](https://tools.ietf.org/html/rfc5280#section-4.2.1.12)], and have a RSA public key length of 2048 bits or higher.
+
+These certificates can be obtaied from a public CA like Symmantec.com, DigiCert.com, GlobalSign.com or Certum.pl 
+> [!Tip]
+> Certum has special offers for [open source projects](https://www.certum.eu/certum/cert,offer_en_open_source_cs.xml).
+
+## Signing with NuGet.exe
+
+NuGet packages can be signed with `NuGet.exe` version 4.6 or higher using the `sign` command. Typically this command will require the next input parameters:
+- **Certificate** to sign with. It can be specified from a PFX file, or from a certificate already installed in the certificate store.
+- **Package** to be signed. It's the nupkg file produced with the `pack` command. If it's already signed it will produce an error unless the `-overwrite` option is used.
+- **Timestamp** server. Optionally you can specify the URL of a timestamp server who implements [RFC 3161](https://tools.ietf.org/html/rfc3161).
+
+```
+nuget sign MyPackage.nupkg -CertificateSubjectName <MyCertSubjectName> -Timestamper http://timestamp.digicert.com 
+```
+
+> [!Note]
+> In this example we are using the digicert.com free timestamp service, but you can use any other service that satisfies the timestamp requirements defined [here](https://github.com/NuGet/Home/wiki/Package-Signatures-Technical-Details).
+
+
+More details on how to use the `sign` command can be found in the [command line reference documentation](#).
+
+> [!Note]
+> Signed packages can be verified with the `verify -signatures` command.
+
+# How to Repository-Sign a package
+
+Repositories interested to add repository signatures to their packages should implement the requirements specified in the [Repository Signatures](https://github.com/NuGet/Home/wiki/Repository-Signatures) specification. The signing can be done using the NuGet api.
```

Comment:
Start with "Repositories that want to add..."

"api" >> "API" and provide a link to the relevant API section.